edb/common/typeutils.py

```diff
@@ -61,3 +61,11 @@ def is_container(obj):
 
 def is_container_type(type_):
     return isinstance(type_, type) and _is_container_type(type_)
+
+
+def dedup(old: Collection[T]) -> List[T]:
+    new: List[T] = []
+    for x in old:
```

Comment:
Since this looks like a generic helper, there will be temptations to use it for larger collections, so maybe use an `OrderedSet` here?  Or just use it directly at call sites?

Reply:
For some reason I had in my head that the `dedup` implementation I put in here used a `set` to track whether things had been seen while building up an ordered result, though obviously it did not.  I'll just use an OrderedSet.